src/main/java/controller/AdminUserController.java

```diff
@@ -0,0 +1,88 @@
+package controller;
+
+import exception.ItemDoesNotExistException;
+import exception.UserDoesNotExistException;
+import exception.UserIsNotAClientUserException;
+import exception.UserIsNotAdminException;
+import item.model.Item;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import request.model.AdminBaseRequest;
+import request.model.AdminItemRequest;
+import users.model.Admin;
+import users.model.ClientUser;
+import users.model.User;
+import users.model.UsersMap;
+
+
+@RestController
+public class AdminUserController {
+
+  @PostMapping(produces = "application/json", value = "/blockUser")
+  public void blockUser(@RequestBody AdminBaseRequest adminBaseRequest) {
+    Admin admin = getAndValidateAdminUser(adminBaseRequest);
+    admin.changeUserStatus(getClientUserAndValidateIt(adminBaseRequest),true);
+  }
+
+  @PostMapping(produces = "application/json", value = "/unblockUser")
+  public void unblockUser(@RequestBody AdminBaseRequest adminBaseRequest) {
+    Admin admin = getAndValidateAdminUser(adminBaseRequest);
+    admin.changeUserStatus(getClientUserAndValidateIt(adminBaseRequest),false);
+  }
+
+  @PostMapping(produces = "application/json",value = "/blockItem")
+  public void blockItem(@RequestBody AdminItemRequest adminItemRequest) {
+    Admin admin = getAndValidateAdminUser(adminItemRequest);
```

Comment:
admin will be used only if an item is found, right?
so declare it inside the if statement